src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -3,12 +3,55 @@
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
 import core.basesyntax.model.User;
+import java.util.Objects;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    public static final int VALID_LOGIN_LENGTH = 6;
+    public static final int VALID_PASSWORD_LENGTH = 6;
+    public static final int VALID_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        Objects.requireNonNull(user);
+        checkUserAlreadyExist(user);
+        checkLoginValid(user);
```

Comment:
this sequence is incorrect

ideally you should verify that login is valid before checking whether it is duplicated

```suggestion
        checkLoginValid(user);
        checkUserAlreadyExist(user);
```